File: RedSheet.DbEntities/Models/vProjectNegotiationRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vProjectNegotiationRecords", Schema = "dbo")]
	[Serializable]
	public partial class vProjectNegotiationRecord  
	{
 
	[Required]
	public System.DateTime EventDate { get; set; } 
 
	[Required] 
	[MaxLength(400)]
	public string IntangiblePowerPlan { get; set; } 
 
	[MaxLength(500)]
	public string KnowAboutThem { get; set; } 
 
	[MaxLength(750)]
	public string KnownIssues { get; set; } 
 
	[Required] 
	[MaxLength(50)]
	public string Location { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int ProjectModuleId { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Range(1, int.MaxValue)]
	public int ProjectNegotiationId { get; set; } 

	public vProjectNegotiationRecord()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-project-negotiation-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vProjectNegotiationRecord {
    constructor(vProjectNegotiationRecord?: vProjectNegotiationRecord )  {
        let properties = [ "eventDate", "intangiblePowerPlan", "knowAboutThem", "knownIssues", "location", "projectModuleId", "projectNegotiationId",];
        for (let property of properties)
            if (vProjectNegotiationRecord && vProjectNegotiationRecord[property])
                this[property] = vProjectNegotiationRecord[property];
    }
 
    @required()
	eventDate : Date =   undefined;
 
    @required()
    @maxLength(400)
	intangiblePowerPlan : string =   undefined;
 
    @maxLength(500)
	knowAboutThem : string =   undefined;
 
    @maxLength(750)
	knownIssues : string =   undefined;
 
    @required()
    @maxLength(50)
	location : string =   undefined;
 
	projectModuleId : number =   0 ;
 
    @range(1,2147483647)
	projectNegotiationId : number =   undefined;


}
